### PR TITLE
Add common static Num's

### PR DESCRIPTION
**Pull request by nimo23**
Changes proposed in this pull request:
- provide common static Num's for `DecimalNum` and `DoubleNum`
- provide access to those Num's in `Num`-Interface

- [x] added an entry with related ticket number(s) to the unreleased section of `CHANGES.md`

### CODE REVIEW

**Comment by nimo23** [comment]
If this PR is taken, then I will add those **static** Num's to all other those needed classes, e.g. the classes described in https://github.com/ta4j/ta4j/issues/731.

The static access may be 
a) faster (we save the time for initializiation)
b) more memory friendly (because only one instance of a static Num is created)

**Comment by nimo23** [comment]
There are 3 alternatives  to the above implementation:

1) move all those static Nums (from `DoubleNum.class`, `DecimalNum.class`, `Num.class`) to enum `NumOf.class` (so that `DoubleNum.class`, `DecimalNum.class`, `Num.class` are unchanged). All static Nums are solely accessed via `NumOf.class`.

2) move the methods from `NumOf.class` to `Num.class` and **remove** `NumOf.class`. All static Nums are accessed via `Num.class` (and `DoubleNum.class`and `DecimalNum.class`).

3) use `NumOf.class` and provide static shortcut methods within `Num.class` (this is the current implemetation as defined in this initial PR). 

Which way should I choose?

**Comment by nimo23** [comment]
Another alternative is to move all methods from `NumOf.class` to `Num.class` and delete `NumOf.class` so that all access to the static Num's can be done through `Num.class` (for example, `Num.HUNDRED(otherNum);`) 

Would it be better to remove `NumOf.class` and move the static methods into `Num.class`?

**Review by team172011 — CHANGES_REQUESTED** [review]
This hardcoded `NumOf` methods depend on the currently available `Num` implementations. If the user would try to use his own/another implementation of `Num` everything calculation would break because `NumOf`... would always return `NaN`.

**Comment by team172011** [comment]
> Would it be better to remove NumOf.class and move the static methods into Num.class?

Yes, because if we want to cache common values we have to do it in a way that it is not bound to the current Num implementations but can be enhanced when someone tries to add his own implementation of Num

**Comment by johnthetireless** [comment]
I am thinking of making all the changes identified in the list of "numOf() offenders" I made for #731   This should use whatever you come up with the end.

Let me know when you have finalized a mechanism the indicators should use and I will fix the ones I listed.

**Comment by nimo23** [comment]
Ok, now it's perfect. I added 

-`Num#zero`
-`Num#one`
-`Num#hundred`

which obtains its value by a static field.

@team172011 any other suggestions? If not, I will go further and add those access to all dedicated classes. Actually, I did the improvement only for `CCIIndicator`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -89,6 +89,7 @@ Changelog for `ta4j`, roughly following [keepachangelog.com](http://keepachangel
 - :tada: **Enhancement** added Position#hasProfit.
 - :tada: **Enhancement** added Position#hasLoss.
 - :tada: **Enhancement** exposed both EMAs in MACD indicator
+- :tada: **Enhancement** added Num#zero, Num#one, Num#hundred.
```
Comment (team172011) [inline CHANGELOG.md]: please update

File: ta4j-core/src/main/java/org/ta4j/core/num/Num.java [hunk 2]
```diff
@@ -41,6 +41,21 @@
...
+
+    /**
+     * @return the Num of 1
+     */
+    Num one();
+
+    /**
+     * @return the Num of 100
+     */
+    public Num hundred();
```
Comment (team172011) [inline ta4j-core/src/main/java/org/ta4j/core/num/Num.java]: Please remove `public` modifier from interface.


we could add a `default` implementation to those methods like

```java
default Num hundred(){
    valueOf(100);
}
```

so once is not forced to override this methods and the implementation would still work. Using this defaults we could remove the implementation from `NaN` class

File: ta4j-core/src/main/java/org/ta4j/core/num/Num.java [hunk 3]
```diff
@@ -41,6 +41,21 @@
  */
 public interface Num extends Comparable<Num>, Serializable {
 
+    /**
+     * @return the Num of 0
+     */
+    Num zero();
```
Comment (team172011) [inline ta4j-core/src/main/java/org/ta4j/core/num/Num.java]: we could enhance javadoc to explain that this is used for caching common used constants

Comment (nimo23) [reply ta4j-core/src/main/java/org/ta4j/core/num/Num.java]: The javadoc for interface should not contain implementation details. But if you want you can add a hint.

**Comment by nimo23** [comment]
Ok, I am done. 

After this PR is merged, another PR can adapt all sources (indicators, etc) where those static Num's can be used.  See the classes defined in https://github.com/ta4j/ta4j/issues/731#issue-863393765.

I will not add those changes in this PR. I recently have not much time. 

@team172011 or  @johnthetireless  Maybe you  can create a new PR and add those static nums to all those needing classes.

**Review by team172011 — APPROVED** [review]

**Pull request merged**